### PR TITLE
Added NetworkConfigurationDiagnostic rest API + example

### PR DESCRIPTION
**Pull request by irrogozh**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by lmazuel** [comment]
Sorry, was the wrong tab, I never meant to close/open the PR :/

**Comment by MikhailTryakhov** [comment]
@irrogozh  please don't forget to merge to August branch

File: specification/network/resource-manager/Microsoft.Network/stable/2018-06-01/networkWatcher.json [hunk 1]
```diff
@@ -3148,6 +3273,221 @@
...
       "description": "Connection state snapshot."
+    },
+    "NetworkConfigurationDiagnosticParameters": {
+      "description": "Parameters to get network configuration diagnostic.",
+      "required": [
+        "targetResourceId",
+        "queries"
+      ],
+      "properties": {
+        "targetResourceId": {
```
Comment (ravbhatnagar) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-06-01/networkWatcher.json]: what permissions are needed on the target resource to perform the network configuration diagnostic

File: specification/network/resource-manager/Microsoft.Network/stable/2018-06-01/networkWatcher.json [hunk 2]
```diff
@@ -3148,6 +3273,221 @@
...
+        "destinationPort": {
+          "type": "string",
+          "description": "Traffice destination port. Accepted values are '*', port (for example, 3389) and port range (for example, 80-100)."
+        }
+      }
+    },
+    "NetworkConfigurationDiagnosticResponse": {
+      "description": "Results of network configuration diagnostic on the target resource.",
+      "properties": {
+        "results": {
```
Comment (ravbhatnagar) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-06-01/networkWatcher.json]: "value" array is the standard here.

File: specification/network/resource-manager/Microsoft.Network/stable/2018-06-01/networkWatcher.json [hunk 3]
```diff
@@ -1594,7 +1594,132 @@
...
+            "description": "Parameters to get network configuration diagnostic."
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
```
Comment (ravbhatnagar) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-06-01/networkWatcher.json]: For all APIs., Model a "default" response to capture all the error states. The schema of this response should comply with ARM RPC error contract. Check Batch RP API spec for a sample.

File: specification/network/resource-manager/Microsoft.Network/stable/2018-06-01/networkWatcher.json (hunk 2 above)
Comment (irrogozh) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-06-01/networkWatcher.json]: It's not ObjectListPage. Value would be there only in this case...

File: specification/network/resource-manager/Microsoft.Network/stable/2018-06-01/networkWatcher.json (hunk 1 above)
Comment (irrogozh) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-06-01/networkWatcher.json]: Probably, read access on NSG.

File: specification/network/resource-manager/Microsoft.Network/stable/2018-06-01/networkWatcher.json (hunk 3 above)
Comment (irrogozh) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-06-01/networkWatcher.json]: Done, thank you!

**Review by lmazuel — APPROVED** [review]

**Pull request merged**